src/semaphore.ts

```diff
@@ -5,10 +5,8 @@ export class Semaphore {
   private _maxAllowed: number;
   private _permits: number;
   private _queue: Unlock[] = [];
-  private _name: string;
 
-  constructor(name: string; maxAllowed: number = 1) {
-    this._name = name;
+  constructor(maxAllowed: number = 1) {
     this._maxAllowed = maxAllowed;
     this._permits = maxAllowed;
   }
```

Comment:
The code patch introduces a class Semaphore with methods for acquiring and releasing permits. The implementation uses async/await for acquire method and maintains a queue of unlock functions. One potential improvement could be the addition of error handling for the asynchronous calls.

Quality Rate: 7/10